pyro/distributions/util.py

```diff
@@ -170,21 +171,23 @@ def scale_and_mask(tensor, scale=1.0, mask=None):
     :param mask: an optional masking tensor
     :type mask: torch.ByteTensor or None
     """
-    if is_identically_zero(tensor):
-        return tensor
-    if mask is None:
-        if is_identically_one(scale):
+    # TODO: refactor is_identically_one / zero to return a
```

Comment:
Yeah, I suppose we should just return `False` here since this is only being used to return early. Will change.

Reply:
I am bypassing this in the calling function itself, because `is_identically_zero` may be used outside of this use case of avoiding trace ops, so its better for the calling function to deal with the JIT warnings instead.